From e9cd19e4212051ed6cd3df081846d90e8c7875ab Mon Sep 17 00:00:00 2001
From: Lorenzo Zolfanelli <lorenzo.zolfanelli@gmail.com>
Date: Fri, 27 Mar 2020 18:48:48 +0100
Subject: [PATCH] fix library and include import for openwrt buildroot

Python helpers in feeds/package/lang/python automatically take cares
of setting proper paths to the gcc cross-compiler.

The custom implementation of BuildExtCommand in python3-lxc would
break this automatic process leading to a build failure, because
it would add the system /usr/include to the gcc standard includes.
---
 setup.py | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/setup.py b/setup.py
index 03b426c..48b112f 100644
--- a/setup.py
+++ b/setup.py
@@ -47,41 +47,6 @@ for var in ("LDFLAGS", "CFLAGS"):
 from setuptools import setup, Extension
 from setuptools.command.build_ext import build_ext as BuildExtCommand
 
-
-class LxcBuildExtCommand(BuildExtCommand):
-    user_options = BuildExtCommand.user_options + [
-        ('no-pkg-config', None,
-         "don't use pkg-config to detect include/library paths")
-    ]
-
-    def initialize_options(self):
-        super(LxcBuildExtCommand, self).initialize_options()
-        self.no_pkg_config = False
-
-    def build_extensions(self):
-        if not self.no_pkg_config:
-            pkg_config_executable = os.environ.get('PKG_CONFIG_EXECUTABLE',
-                                                   'pkg-config')
-
-            def get_pkg_config_var(name):
-                args = [pkg_config_executable, '--variable', name, 'lxc']
-                output = subprocess.check_output(args,
-                                                 universal_newlines=True)
-                return output.rstrip('\n')
-
-            try:
-                includedir = get_pkg_config_var('includedir')
-                libdir = get_pkg_config_var('libdir')
-
-                self.compiler.add_include_dir(includedir)
-                self.compiler.add_library_dir(libdir)
-
-            except subprocess.CalledProcessError:
-                pass
-
-        super(LxcBuildExtCommand, self).build_extensions()
-
-
 setup(name='python3-lxc',
       version='3.0.4',
       description='Python3 bindings for LXC',
@@ -93,5 +58,4 @@ setup(name='python3-lxc',
       packages=['lxc'],
       package_dir={'lxc': 'lxc'},
       ext_modules=[Extension('_lxc', sources=['lxc.c'], libraries=['lxc'])],
-      cmdclass={'build_ext': LxcBuildExtCommand},
       )
-- 
2.26.0

